implementations/graphql-dotnet/Program.cs

```diff
@@ -1,111 +1,37 @@
-                        if (user != null)
-                        {
-                            string? totalProductsCreated = ctx.Arguments.GetValueOrDefault("totalProductsCreated")?.ToString();
-                            if (totalProductsCreated != null)
-                            {
-                                user = user with { TotalProductsCreated = int.Parse(totalProductsCreated) };
-                            }
-
-                            string? yearsOfEmployment = ctx.Arguments.GetValueOrDefault("yearsOfEmployment")?.ToString();
-                            if (yearsOfEmployment != null)
-                            {
-                                user.LengthOfEmployment = int.Parse(yearsOfEmployment);
-                            }
-                        }
-                        return Task.FromResult(user);
-                    }
-
-                    return Task.FromResult<User?>(null);
-                });
-            schemaBuilder.Types.For(nameof(ProductResearch))
-                .ResolveReferenceAsync<ProductResearch?>(ctx =>
-                {
-                    var study = ctx.Arguments.GetValueOrDefault("study");
-                    if (study is Dictionary<string, object>)
-                    {
-                        var caseNumber = ((Dictionary<string, object>)study).GetValueOrDefault("caseNumber")?.ToString();
-                        return Task.FromResult(Data.ProductResearches.FirstOrDefault(r => r.Study.CaseNumber == caseNumber));
-                    }
-
-                    return Task.FromResult<ProductResearch?>(null);
-                });
-        });
-    })
-    .AddGraphQL(graphqlBuilder =>
-    {
-
-        graphqlBuilder.UseApolloTracing();
-        // graphqlBuilder.ConfigureExecutionOptions(action =>
-        // {
-        //     action.UnhandledExceptionDelegate =
-
-        // });
-        graphqlBuilder.AddSystemTextJson();
-        graphqlBuilder.AddErrorInfoProvider(action =>
+            link.Imports.Add("@interfaceObject", "@interfaceObject");
+            link.Imports.Add("@composeDirective", "@composeDirective");
+            link.Imports.Add("@extends", "@extends");
+        })
+        .ConfigureSchema(schema =>
         {
-            action.ExposeExceptionDetails = true;
-        });
-    });
+            schema.Directives.Register(new Directive("custom", DirectiveLocation.Object));
+            schema.LinkSchema("https://myspecs.dev/myCustomDirective/v1.0", link => link.Imports.Add("@custom", "@custom"));
+            schema.ApplyDirective("composeDirective", "name", "@custom");
+            schema.RegisterType<AutoRegisteringObjectGraphType<Inventory>>();
+        })
+        .ConfigureExecutionOptions(options => options.ThrowOnUnhandledException = true));
 
 var app = builder.Build();
 
-app.UseCors();
+// Sample: print the schema to a file
+File.WriteAllText("products.graphql", app.Services.GetRequiredService<ISchema>().Print());
```

Comment:
So that the sdl here is the same as what is exposed with `{ _service { sdl } }`:

```suggestion
File.WriteAllText("products.graphql", app.Services.GetRequiredService<ISchema>().Print(new FederationPrintOptions()));
```